src/basic/io-util.c

```diff
@@ -306,3 +318,18 @@ ssize_t sparse_write(int fd, const void *p, size_t sz, size_t run_length) {
 
         return q - (const uint8_t*) p;
 }
+
+_noreturn_ void report_errno_and_exit(int errno_fd, int r) {
+        if (r >= 0)
+                _exit(EXIT_SUCCESS);
+
+        assert(errno_fd >= 0);
+
+        int rr = loop_write_exact(errno_fd, &r, sizeof(r));
+        if (rr == -EIO)
+                log_debug_errno(SYNTHETIC_ERRNO(EIO), "Sent unexpectedly short message");
+        if (rr < 0)
+                log_debug_errno(rr, "Failed to write errno to errno_fd=%d: %m", errno_fd);
```

Comment:
this would result in duplicate logging on EIO.

i.e. an `else` is missing here